### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by Xavrax**
test: add unit tests for domain modules

add unit tests for domain modules

### CODE REVIEW

File: src/domain/parsers.cpp [hunk 1]
```diff
@@ -126,8 +126,8 @@ std::pair<Parsers::PubnubJson::Timetoken, MessageEntity> Parsers::PubnubJson::to
             {Pubnub::MessageAction{
             Pubnub::message_action_type_from_string(json_field_from_pn_block(pn_message.payload, "data", "type")),
                 json_field_from_pn_block(pn_message.payload, "data", "value"),
-                json_field_from_pn_block(pn_message.payload, "data", "uuid"),
-                json_field_from_pn_block(pn_message.payload, "data", "actionTimetoken")
+                json_field_from_pn_block(pn_message.payload, "data", "actionTimetoken"),
+                json_field_from_pn_block(pn_message.payload, "data", "uuid")
```
Comment (KGronek-Pubnub) [inline src/domain/parsers.cpp:130]: Why is the order changed, does it matter?

File: src/domain/typing.cpp [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #include "typing.hpp"
 #include "domain/json.hpp"
+#include "string.hpp"
+#include <iostream>
```
Comment (KGronek-Pubnub) [inline src/domain/typing.cpp:4]: Do we need iostream or it's a leftover from debugging?

**Review by KGronek-Pubnub — APPROVED** [review]
LGTM

File: src/domain/typing.cpp (hunk 2 above)
Comment (Xavrax) [reply src/domain/typing.cpp:4]: I will remove it.

**Pull request merged**